File: c-sharp/Models/CaesarText.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Ciphers.Models
{
    public class CaesarText : Text
    {
        public int Key { get; set; }

        public CaesarText(string plainTxt, int key) : base(plainTxt)
        {
            Key = key;
        }
    }
}

File: java/src/main/java/com/metatarsal/ciphers/models/CaesarText.java
package com.metatarsal.ciphers.models;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode()
public class CaesarText extends Text {

    private int key;

    public CaesarText(String plainText, int key) {
        super();
        this.plainText = plainText;
        this.key = key;
    }

    public int getKey() {
        return key;
    }
}
